src/node_url.cc

```diff
@@ -1267,30 +1297,37 @@ void URL::Parse(const char* input,
     len = end - p;
   }
 
+  // The spec says we should strip out any ASCII tabs or newlines.
+  // In those cases, we create another std::string instance with the filtered
+  // contents, but in the general case we avoid the overhead.
   std::string whitespace_stripped;
-  whitespace_stripped.reserve(len);
-  for (const char* ptr = p; ptr < end; ptr++)
+  for (const char* ptr = p; ptr < end; ptr++) {
     if (!IsASCIITabOrNewline(*ptr))
-      whitespace_stripped += *ptr;
+      continue;
+    // Hit tab or newline. Allocate storage, copy what we have until now,
+    // and then iterate and filter all similar characters out.
+    whitespace_stripped.reserve(len - 1);
+    whitespace_stripped.assign(p, ptr - p);
+    // 'ptr + 1' skips the current char, which we know to be tab or newline.
+    for (ptr = ptr + 1; ptr < end; ptr++) {
+      if (!IsASCIITabOrNewline(*ptr))
+        whitespace_stripped += *ptr;
+    }
 
-  input = whitespace_stripped.c_str();
-  len = whitespace_stripped.size();
-  p = input;
-  end = input + len;
+    // Update variables like they should have looked like if the string
+    // had been stripped of whitespace to begin with.
+    input = whitespace_stripped.c_str();
+    len = whitespace_stripped.size();
+    p = input;
+    end = input + len;
+    break;
+  }
 
-  bool atflag = false;
-  bool sbflag = false;
-  bool uflag = false;
+  bool atflag = false;  // Set when @ has been seen.
+  bool square_bracket_flag = false;  // Set inside of [...]
+  bool inside_password_flag = false;  // Set after a : after an username.
```

Comment:
As a data point, the URL Standard calls it passwordTokenSeenFlag, but seems like you figured out what it meant on yourself :)

Reply:
@TimothyGu Idk, do you want me to change it? I think both of these are okay names?

Reply:
I'd slightly prefer the name in the spec for better correspondence for future readers.